test/e2e/flask/solana-wallet-standard/connect.spec.ts

```diff
@@ -0,0 +1,358 @@
+      await withSolanaAccountSnap(
+        {
+          ...DEFAULT_SOLANA_TEST_DAPP_FIXTURE_OPTIONS,
+          title: this.test?.fullTitle(),
+        },
+        async (driver) => {
+          const testDapp = new TestDappSolana(driver);
+          await testDapp.openTestDappPage();
+          await connectSolanaTestDapp(driver, testDapp);
+
+          const header = await testDapp.getHeader();
+          const account = await header.getAccount();
+          assertConnected(account, account1Short);
+
+          await driver.refresh();
+
+          const accountAfterRefresh = await header.getAccount();
+          assertConnected(accountAfterRefresh, account1Short);
+        },
+      );
+    });
+
+    describe('Given I have connected to two accounts', function () {
+      it('Refreshing the page should keep me connected to the last selected account', async function () {
+        await withSolanaAccountSnap(
+          {
+            ...DEFAULT_SOLANA_TEST_DAPP_FIXTURE_OPTIONS,
+            title: this.test?.fullTitle(),
+            numberOfAccounts: 2,
+          },
+          async (driver) => {
+            const testDapp = new TestDappSolana(driver);
+            await testDapp.openTestDappPage();
+            await connectSolanaTestDapp(driver, testDapp, {
+              selectAllAccounts: true,
+            });
+
+            await driver.refresh();
+
+            const header = await testDapp.getHeader();
+            const account = await header.getAccount();
+            assertConnected(account, account2Short);
+          },
+        );
+      });
+    });
+
+    describe('Given I have connected to Mainnet and Devnet', function () {
+      it('Should use the Mainnet scope by default', async function () {
+        await withSolanaAccountSnap(
+          {
+            ...DEFAULT_SOLANA_TEST_DAPP_FIXTURE_OPTIONS,
+            title: this.test?.fullTitle(),
+          },
+          async (driver) => {
+            const testDapp = new TestDappSolana(driver);
+            await testDapp.openTestDappPage();
+            await connectSolanaTestDapp(driver, testDapp, {
+              includeDevnet: true,
+            });
+
+            // Refresh the page
+            await driver.refresh();
+
+            // Set the endpoint to devnet as it has been reset after the refresh
+            const header = await testDapp.getHeader();
+            await header.setEndpoint(SOLANA_DEVNET_URL);
+            await driver.clickElement({ text: 'Update', tag: 'button' });
+
+            await driver.delay(regularDelayMs);
+
+            const signMessageTest = await testDapp.getSignMessageTest();
+            await signMessageTest.setMessage('Hello, world!');
+            await signMessageTest.signMessage();
+
+            await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+            // Check that devnet appears in the dialog
+            const el = await driver.findElement(
+              By.xpath("//p[text()='Solana Mainnet']"),
```

Comment:
Atleast this seems to be what the test is sayin? I'm not sure I follow the reason for this UX yet though
```suggestion
            // Check that mainnet appears in the dialog
            const el = await driver.findElement(
              By.xpath("//p[text()='Solana Mainnet']"),
```